### PR TITLE
Fix `RoleDescriptor` field annotations for Serverless

### PR DESCRIPTION
**Pull request by n1v0lg**
I included these changes for `SecurityPutRoleRequest` but they also need to be part of the role descriptor model since that is referenced by e.g. the API key APIs.

### CODE REVIEW

**Comment by pquentin** [comment]
Note that you don't need to click on the "Update branch" button whenever the branch is out-of-date, especially as today is a busy day regarding merges.

File: specification/security/put_role/SecurityPutRoleRequest.ts [hunk 1]
```diff
@@ -72,6 +72,10 @@ export interface Request extends RequestBase {
      * @doc_id run-as-privilege
      */
     run_as?: string[]
+    /**
+     * Optional description of the role descriptor
+     */
+    description?: string
```
Comment (n1v0lg) [inline specification/security/put_role/SecurityPutRoleRequest.ts:78]: This was missing, and leading to validation failures so drive-by fixing it.

**Comment by n1v0lg** [comment]
Note: RoleDescriptor is missing `remote_indices` and `remote_cluster` field definitions -- I will add these in a follow up PR.

**Review by pquentin — APPROVED** [review]
Excellent, thank you!

**Comment by pquentin** [comment]
Backport:
 * https://github.com/elastic/elasticsearch-specification/pull/2702

**Pull request merged**